user/applications/tinker/src/burnin_test.cpp

```diff
@@ -568,25 +579,40 @@ bool BurninTest::testGnss() {
 
     // Enable GNSS. It can take some time after the modem AT interface comes up for the GNSS engine to start
     const int RETRIES = 10;
-    int r = 0;
-    for (int i = 0; i < RETRIES && r != RESP_OK; i++) {
-        r = Cellular.command("AT+QGPS=1\r\n");
+    
+    bool success = false; 
+    for (int i = 0; i < RETRIES && !success; i++) {
+        Cellular.command(callbackQGPS, &success, 5000, "AT+QGPS=1");
         delay(1000);
     }
 
-    if (r != RESP_OK) {
-        strcpy(BurninErrorMessage, "AT+QGPS=1 failed, GNSS not enabled");
+    if (!success) {
+        Log.error("AT+QGPS=1 failed, GNSS not enabled");
         return false;
     }
 
-    // Configure antenna for GNSS priority
-    for (int i = 0; i < RETRIES && r != RESP_OK; i++) {
-        r = Cellular.command("AT+QGPSCFG=\"priority\",0");
-        delay(1000);
+    hal_device_hw_info deviceInfo = {};
+    hal_get_device_hw_info(&deviceInfo, nullptr);
+    if (deviceInfo.ncp[0] == PLATFORM_NCP_QUECTEL_BG95_M5) {
+        int r = 0;
+        // Configure antenna for GNSS priority
+        for (int i = 0; i < RETRIES && r != RESP_OK; i++) {
+            r = Cellular.command("AT+QGPSCFG=\"priority\",0");
```

Comment:
The `AT+QGPSCFG="priority"` command should take a maximum of 300ms to execute.  I see you don't have a timeout on the command call.

Reply:
Right, I just use the default 10 second timeout. I dont care about timing just that it succeeded.